### DISCUSSION TITLE
Maximum jobs per user

### DISCUSSION THREAD

**Discussion by kevintprod** [taskforcesh/bullmq#1461] [Q&A]
Hi Everyone,

I did read the document for this, I even have the codes that were running on my local. Now, I can't figure out how I implement jobs per user. I am developing an app that cleans an email if the email is valid and I am using MySQL for it as background jobs.

My idea is to limit the jobs if ever the user already running a job that has 1M rows in MySQL.

Anyone?

**Comment by EmanuelCampos** [comment]
> Hi Everyone,
> 
> I did read the document for this, I even have the codes that were running on my local. Now, I can't figure out how I implement jobs per user. I am developing an app that cleans an email if the email is valid and I am using MySQL for it as background jobs.
> 
> My idea is to limit the jobs if ever the user already running a job that has 1M rows in MySQL.
> 
> Anyone?

why not store a count on your database and then block him when the limit is reached?

**Reply by kevintprod** [reply]
Yes, I am using the database to count if ever the limit is reached.

Right now, I am adding a job to "Waiting" queue and evaluating if the user has reached the limit so the job can stay to Waiting queue. and once a job is complete all Waiting needs to re-evaluate if ever the user can proceed to activate the job.

Is there a method for job like "moveToWaitingQueue"?

**Comment by manast** [comment]
Maybe what you need is groups: https://docs.bullmq.io/bullmq-pro/groups
